opsdroid/connector/matrix/connector.py

```diff
@@ -223,20 +247,35 @@ async def get_nick(self, roomid, mxid):
         setting.
         """
         if self.room_specific_nicks:
-            try:
-                return await self.connection.get_room_displayname(roomid, mxid)
-            except Exception:  # pylint: disable=W0703
-                # Fallback to the non-room specific one
-                logging.exception("Failed to lookup room specific nick for %s.", mxid)
+            if roomid is None:
+                return mxid
+
+            res = await self.connection.joined_members(roomid)
+            if isinstance(res, nio.JoinedMembersError):
+                logging.exception("Failed to lookup room members for %s.", roomid)
+                # fallback to profile
+                res = await self.connection.get_displayname(mxid)
+                if isinstance(res, nio.ProfileGetDisplayNameError):
+                    _LOGGER.error("Failed to lookup nick for %s.", mxid)
+                    return mxid
+                if res.displayname is None:
+                    return mxid
+                return res.displayname
+
+            for member in res.members:
+                if member.user_id == mxid:
+                    return member.display_name
```

Comment:
I think you can re-write this logic so that it drops out of the top `if self.room_specific_nicks` block to resolve the global profile nick, rather than rewriting this here.

```suggestion
            else:
                for member in res.members:
                    if member.user_id == mxid:
                        return member.display_name
```

Reply:
Anyone has an idea why the one test fails? Seems like the error is not logged but I have no clue why